frontends/web/src/routes/new-settings/components/settingsItemContainer/settingsItemContainer.tsx

```diff
@@ -0,0 +1,31 @@
+import { ReactNode } from 'react';
+import styles from './settingsItemContainer.module.css';
+
+interface SettingsButtonProps {
+    onClick?: () => void;
+    settingName: string;
+    secondaryText?: string | JSX.Element;
+    extraComponent?: ReactNode;
+}
+
+export const SettingsItemContainer = ({
+  onClick,
+  settingName,
+  secondaryText,
+  extraComponent
+}: SettingsButtonProps) => {
+  const notButton = onClick === undefined;
+  return (
+    <button
+      className={`${styles.container} ${notButton ? `${styles.notButton}` : ''}`}
+      onClick={onClick}>
+      <span>
+        <p className={styles.primaryText}>{settingName}</p>
+        { secondaryText ? (
+          <p className={styles.secondaryText}>{secondaryText}</p>
+        ) : null }
+      </span>
+      {extraComponent ? extraComponent : null}
+    </button>
```

Comment:
> This will render other input/select elements inside a button element.

Oh good point..

> Can you change so the container is a <div> if there is an extraComponent or make a 2nd component for non-buttons WDYT?

Will do!